bulk_handlers_test.go

```diff
@@ -244,6 +245,69 @@ func TestBulkCreate(t *testing.T) {
 	}
 }
 
+func TestBulkCreateSourceValidationBadRequest(t *testing.T) {
+	testutils.SkipIfNotRunningIntegrationTests(t)
+	tenantId := int64(1)
+
+	// Create a source
+	reqParams := dao.RequestParams{TenantID: &tenantId}
+	sourceDao := dao.GetSourceDao(&reqParams)
+
+	uid := "bd2ba6d6-4630-40e2-b829-cf09b03bdb9f"
+	nameSource := "Source for TestBulkCreateSourceValidationBadRequest()"
+	src := m.Source{
+		Name:         nameSource,
+		SourceTypeID: 1,
+		Uid:          &uid,
+	}
+
+	err := sourceDao.Create(&src)
+	if err != nil {
+		t.Errorf("source not created correctly: %s", err)
+	}
+
+	// Try to create same source via bulk create
+	sourceTypeName := "bitbucket"
+	applicationTypeName := "app-studio"
+	authenticationResourceType := "application"
+
+	requestBody := testutils.SingleResourceBulkCreateRequest(nameSource, sourceTypeName, applicationTypeName, authenticationResourceType)
+
+	body, err := json.Marshal(requestBody)
+	if err != nil {
+		t.Error("Could not marshal JSON")
+	}
+
+	c, res := request.CreateTestContext(
+		http.MethodPost,
+		"/api/sources/v3.1/bulk_create",
+		bytes.NewReader(body),
+		map[string]interface{}{
+			"tenantID": tenantId,
+		},
+	)
+
+	c.Request().Header.Add("Content-Type", "application/json;charset=utf-8")
+	c.Set("identity", &identity.XRHID{Identity: identity.Identity{AccountNumber: fixtures.TestTenantData[0].ExternalTenant}})
+
+	badRequestBulkCreate := ErrorHandlingContext(BulkCreate)
+	err = badRequestBulkCreate(c)
+	if err != nil {
+		t.Error(err)
+	}
+
+	templates.BadRequestTest(t, res)
+
+	// Delete created source
+	deletedSource, err := sourceDao.Delete(&src.ID)
```

Comment:
do we need `cleanSourceForTenant` or sourceDAO#Delete handles run also DeleteCascade ? (I guess there were created more objects in DB with using bulk create )

Reply:
I believe we don't because in this test only one source is created via simple source create and then I try to create the same source via bulk create but this attempt leads to bad request err because this source already exists ... so that why is not needed the cascade delete

Reply:
I see, thanks!